### ISSUE TITLE
Unexpected behaviour during Amplify.Auth.signOut()  after Amplify.Auth.signInWithWebUI()

### ISSUE THREAD

**Issue by NepipenkoIgor** [aws-amplify/amplify-flutter#1452]
### Description

I need call signOut twice because  first time a see web page without actions .
<img width="442" alt="Screenshot 2022-03-11 at 12 29 48" src="https://user-images.githubusercontent.com/7725146/157850169-791905d4-488f-4903-ab5e-26d9d7d1fda7.png">
I can click only  cancel and after this i got an error 
```
Unhandled Exception: UserCancelledException(message: User cancelled the signOut flow and could not be completed., recoverySuggestion: Present the signOut UI again for the user to sign out., underlyingException: The operation couldn’t be completed. (AmplifyPlugins.AWSCognitoAuthError error 19.))
```
then i try call again  signOut and everything will be ok 

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

All steps in description 

### Screenshots

Provided in description

### Platforms

- [X] iOS
- [X] Android

### Environment

```bash
✗ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.2.1 21D62 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.64.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

### Dependencies

```bash
flutter pub deps --no-dev --style=compact
Dart SDK 2.16.1
Flutter SDK 2.10.3
core_workspace 0.4.3

dependencies:
- core 0.4.3 [flutter flutter_screenutil flutter_svg amplify_auth_cognito amplify_flutter provider dio sentry_flutter package_info_plus connectivity_plus flutter_blue shared_preferences flutter_markdown ui_kit]
- flutter 0.0.0 [characters collection material_color_utilities meta typed_data vector_math sky_engine]

transitive dependencies:
- amplify_analytics_plugin_interface 0.4.1 [amplify_core flutter meta]
- amplify_api_plugin_interface 0.4.1 [amplify_core collection flutter json_annotation meta]
- amplify_auth_cognito 0.4.1 [flutter amplify_auth_plugin_interface amplify_core collection plugin_platform_interface]
- amplify_auth_plugin_interface 0.4.1 [flutter meta amplify_core]
- amplify_core 0.4.1 [flutter plugin_platform_interface collection date_time_format meta uuid]
- amplify_datastore_plugin_interface 0.4.1 [flutter meta collection amplify_core]
- amplify_flutter 0.4.1 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interface amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- amplify_storage_plugin_interface 0.4.1 [flutter meta amplify_core]
- animated_stack_widget 0.0.4 [flutter]
- args 2.3.0
- async 2.8.2 [collection meta]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.15.0
- connectivity_plus 2.2.1 [flutter connectivity_plus_platform_interface connectivity_plus_linux connectivity_plus_macos connectivity_plus_web connectivity_plus_windows]
- connectivity_plus_linux 1.3.0 [flutter connectivity_plus_platform_interface meta nm]
- connectivity_plus_macos 1.2.1 [connectivity_plus_platform_interface flutter]
- connectivity_plus_platform_interface 1.2.0 [flutter meta plugin_platform_interface]
- connectivity_plus_web 1.2.0 [connectivity_plus_platform_interface flutter_web_plugins flutter]
- connectivity_plus_windows 1.2.0 [connectivity_plus_platform_interface flutter]
- convert 3.0.1 [typed_data]
- crypto 3.0.1 [collection typed_data]
- date_time_format 2.0.1
- dbus 0.7.1 [args ffi meta xml]
- dio 4.0.4 [http_parser path]
- ffi 1.1.2
- file 6.1.2 [meta path]
- fixnum 1.0.0
- flutter_blue 0.8.0 [flutter convert protobuf rxdart collection meta]
- flutter_map 0.14.0 [async collection flutter http intl latlong2 meta positioned_tap_detector_2 proj4dart transparent_image tuple vector_math]
- flutter_map_marker_cluster 0.4.4 [flutter flutter_map latlong2 flutter_map_marker_popup]
- flutter_map_marker_popup 2.1.2 [flutter flutter_map latlong2 animated_stack_widget]
- flutter_markdown 0.6.9 [flutter markdown meta path]
- flutter_screenutil 5.3.1 [flutter]
- flutter_spinkit 5.1.0 [flutter]
- flutter_svg 1.0.3 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta typed_data vector_math]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- js 0.6.3
- json_annotation 4.4.0 [meta]
- latlong2 0.8.1 [intl]
- lists 1.0.1 [meta]
- location 4.3.0 [flutter location_platform_interface location_web]
- location_platform_interface 2.3.0 [flutter meta plugin_platform_interface]
- location_web 3.1.1 [flutter flutter_web_plugins http_parser js location_platform_interface meta]
- markdown 4.0.1 [args charcode meta]
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.3
- meta 1.7.0
- mgrs_dart 2.0.0 [unicode]
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- package_info_plus 1.4.0 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- package_info_plus_linux 1.0.3 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.4 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.4 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.0
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- path_provider_linux 2.1.5 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.3 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.5 [ffi flutter path path_provider_platform_interface win32]
- petitparser 4.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- positioned_tap_detector_2 1.0.4 [flutter]
- process 4.2.4 [file path platform]
- proj4dart 2.0.0 [mgrs_dart wkt_parser meta]
- protobuf 2.0.1 [fixnum collection]
- provider 6.0.2 [collection flutter nested]
- qr_code_scanner 0.6.1 [js flutter flutter_web_plugins]
- quiver 3.0.1+1 [matcher]
- rxdart 0.26.0
- sentry 6.3.0 [http meta stack_trace uuid]
- sentry_flutter 6.3.0 [flutter flutter_web_plugins sentry package_info_plus]
- shared_preferences 2.0.13 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- shared_preferences_android 2.0.11 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.0 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.3 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.3 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- transparent_image 2.0.0
- tuple 2.0.0 [quiver]
- typed_data 1.3.0 [collection]
- ui_kit 0.0.30 [flutter flutter_spinkit qr_code_scanner flutter_map flutter_map_marker_cluster location]
- unicode 0.3.1 [lists]
- uuid 3.0.6 [crypto]
- vector_math 2.1.1
- win32 2.4.1 [ffi]
- wkt_parser 2.0.0
- xdg_directories 0.2.0+1 [meta path process]
- xml 5.3.1 [collection meta petitparser]
```

### Device

iPhone XR and IPhone 13 Simulator

### OS

iOS 15.1

### CLI Version

no

**Comment by Jordan-Nelson** [comment]
Hi @NepipenkoIgor - Are you seeing this with any other auth providers besides Microsoft/Azure?

**Comment by NepipenkoIgor** [comment]
@Jordan-Nelson  Can't answer very fast. Currently i don't have any another providers . I will be able to check in next days when backend team will create additional configuration to another provider

**Comment by Jordan-Nelson** [comment]
Okay. If you are seeing this for other auth providers (google, apple), I would guess that something isn't configured correctly. We can attempt to reproduce this, but it may take a bit since it will require setting up a new auth provider.

Let me know if you end up seeing this for other providers.

**Comment by NepipenkoIgor** [comment]
@Jordan-Nelson  Hi again. Just checked with google provider everything good . No such behaviour that i described  for Azure.

**Comment by NepipenkoIgor** [comment]
@Jordan-Nelson  Also short question for IOS  when i call Amplify.Auth.signInWithWebUI() i always have this confirmation window
<img width="378" alt="Screenshot 2022-03-11 at 20 28 29" src="https://user-images.githubusercontent.com/7725146/157929717-3b3976ce-0c80-4d09-8532-2a2ab6180802.png">

For android no confirm window and this is what i am expect . Could i skip this confirmation for IOS. Maybe you know how to manage this ?

**Comment by Jordan-Nelson** [comment]
> Could i skip this confirmation for IOS

@NepipenkoIgor - You can use [isPreferPrivateSession](https://pub.dev/documentation/amplify_auth_cognito/latest/amplify_auth_cognito/CognitoSignInWithWebUIOptions/isPreferPrivateSession.html) to remove it. Setting this to true will set [prefersEphemeralWebBrowserSession](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/3237231-prefersephemeralwebbrowsersessio?language=objc) true on iOS.

You should be aware that this will use a private browser session, which means that any credentials stored in the user's browser will not be available. So if they are already logged into their Google account in Safari, they will be forced to log in again.

> Just checked with google provider everything good . No such behaviour that i described for Azure.

Okay, I can attempt to reproduce this, but it may take a bit since it will involve setting up a new auth provider.

**Comment by NepipenkoIgor** [comment]
@Jordan-Nelson  `isPreferPrivateSession` woks as expected . Thank you.

Also i will wait your feedback about provider signOut behaviour .

**Comment by Ekoios-Teddy** [comment]
> > Could i skip this confirmation for IOS
> 
> @NepipenkoIgor - You can use [isPreferPrivateSession](https://pub.dev/documentation/amplify_auth_cognito/latest/amplify_auth_cognito/CognitoSignInWithWebUIOptions/isPreferPrivateSession.html) to remove it. Setting this to true will set [prefersEphemeralWebBrowserSession](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/3237231-prefersephemeralwebbrowsersessio?language=objc) true on iOS.
> 
> You should be aware that this will use a private browser session, which means that any credentials stored in the user's browser will not be available. So if they are already logged into their Google account in Safari, they will be forced to log in again.
> 
> > Just checked with google provider everything good . No such behaviour that i described for Azure.
> 
> Okay, I can attempt to reproduce this, but it may take a bit since it will involve setting up a new auth provider.

@Jordan-Nelson `isPreferPrivateSession` will be forced to log in again on ios but when login on android device. When only one account google logged in browser, ` Amplify.Auth.signInWithWebUI` auto pass login without re-enter credentials. So i want to always have to re-enter credential after every login. Does this support on android?

**Comment by Jordan-Nelson** [comment]
@Ekoios-Teddy - There is no equivalent of isPreferPrivateSession for Android. There is a request being tracked to support the option to always show the choose account dialog in https://github.com/aws-amplify/amplify-flutter/issues/1528.

**Comment by dnys1** [comment]
We believe this issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. Can someone experiencing this issue confirm that to be the case?

This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
